backend/src/org/commcare/resources/model/ResourceTable.java

```diff
@@ -419,83 +433,64 @@ public void prepareResources(ResourceTable master, CommCareInstance instance)
      * @throws UnfullfilledRequirementsException resource(s) incompatible with
      *                                           current CommCare version
      */
-    public void prepareResources(ResourceTable master,
-                                 CommCareInstance instance,
-                                 String toInitialize)
-            throws UnresolvedResourceException, UnfullfilledRequirementsException {
+    public void prepareResourcesUpTo(ResourceTable master,
```

Comment:
Changing the name to `prepareProfileResource` would make me want to avoid taking in a `toInitialize` argument, which would require the `ResourceTable` to import the profile reference key from `CommCarePlatform`. Importing very  specific info to a generic class like `ResourceTable` feels a little wrong. I'm going to create a helper method in `ResourceManager` with a better name that calls `prepareResourcesUpTo`